pkg/apis/cluster/validation/validation.go

```diff
@@ -157,9 +181,11 @@ func validateClusterSpecForFieldSelector(spec *api.ClusterSpec, fldPath *field.P
 	if errs := kubevalidation.IsValidLabelValue(spec.Region); len(errs) != 0 {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("region"), spec.Region, strings.Join(errs, "; ")))
 	}
-	if errs := kubevalidation.IsValidLabelValue(spec.Zone); len(errs) != 0 {
-		allErrs = append(allErrs, field.Invalid(fldPath.Child("zone"), spec.Zone, strings.Join(errs, "; ")))
+
+	if errs := validateClusterSpecZoneLabel(spec, fldPath); len(errs) != 0 {
+		allErrs = append(allErrs, errs...)
 	}
```

Comment:
Given the `.spec.zone` will be removed eventually, so my suggestion is not couple them together, we could do it like:
```golang
	if spec.Zone != "" && len(spec.Zones) > 0 {
		// return errors
	}
	
	if errs := kubevalidation.IsValidLabelValue(spec.Zone); len(errs) != 0 {
		allErrs = append(allErrs, field.Invalid(fldPath.Child("zone"), spec.Zone, strings.Join(errs, "; ")))
	}
	
	for i := range spec.Zones {
		// validate each zones.
	}
```
That, when we remove the .spec.zone, we can easily remove the relevant code.

Reply:
good suggestion.